streams/test-utils/src/test/java/org/apache/kafka/streams/test/MockFixedKeyProcessorContextTest.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.test;
+
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.Punctuator;
+import org.apache.kafka.streams.processor.TaskId;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessor;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessorContext;
+import org.apache.kafka.streams.processor.api.FixedKeyRecord;
+import org.apache.kafka.streams.processor.api.InternalFixedKeyRecordFactory;
+import org.apache.kafka.streams.processor.api.MockFixedKeyProcessorContext;
+import org.apache.kafka.streams.processor.api.MockProcessorContext.CapturedForward;
+import org.apache.kafka.streams.processor.api.Record;
+import org.apache.kafka.streams.processor.api.RecordMetadata;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.apache.kafka.common.utils.Utils.mkProperties;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+public class MockFixedKeyProcessorContextTest {
```

Comment:
Can you parameterize `MockProcessorContextAPITest`? Most of the code here seems to be the same as what's in `MockProcessorContextAPITest`?

Reply:
Definitely! Do you think I should just add some specific FixedKeyProcessor tests to the MockProcessorContextAPITest file since it is mostly the same?

Reply:
Right. I think you can parameterize `MockProcessorContextAPITest` so common tests run for both processor context mock and add some tests for `FixedKeyProcessor` only (return directly for other processor context)

Reply:
Sounds good, just to make sure I'm on the right track, here is a code snippet
```
    @ParameterizedTest(name = "processor = {0}, context = {1}")
    @MethodSource(value = "parameters")
    @SuppressWarnings("unchecked")
    public void shouldStoreAndReturnStateStores(Object processorObject, ProcessingContext context) {

        Record<Object, Object> input1 = new Record<>("foo", 5L, 0L);
        Record<Object, Object> input2 = new Record<>("bar", 50L, 0L);

        final StoreBuilder<KeyValueStore<String, Long>> storeBuilder = Stores.keyValueStoreBuilder(
            Stores.inMemoryKeyValueStore("my-state"),
            Serdes.String(),
            Serdes.Long()).withLoggingDisabled();

        final KeyValueStore<String, Long> store = storeBuilder.build();
        
        if(processorObject instanceof Processor && context instanceof MockProcessorContext){
            Processor<Object, Object, Object, Object> processor = ((Processor<Object, Object, Object, Object>) processorObject);
            MockProcessorContext<Object, Object> mockProcessorContext = ((MockProcessorContext<Object, Object>) context);
            
            store.init(mockProcessorContext.getStateStoreContext(), store);

            processor.init(mockProcessorContext);
            processor.process(input1);
            processor.process(input2);
        }
        else if(processorObject instanceof FixedKeyProcessor && context instanceof MockFixedKeyProcessorContext){
            FixedKeyProcessor<Object, Object, Object> processor = ((FixedKeyProcessor<Object, Object, Object>) processorObject);
            MockFixedKeyProcessorContext<Object, Object> mockFixedKeyProcessorContext = ((MockFixedKeyProcessorContext<Object, Object>) context);
            
            store.init(mockFixedKeyProcessorContext.getStateStoreContext(), store);
            
            processor.init(mockFixedKeyProcessorContext);
            processor.process(InternalFixedKeyRecordFactory.create(input1));
            processor.process(InternalFixedKeyRecordFactory.create(input2));
        }
        else{
            assert false;
        }
```
        
Since `FixedKeyProcessor` and `Processor` are not related to one another, this would be a way to pass in both processors to one test.